components/webgpu/swapchain.rs

```diff
@@ -309,6 +323,23 @@ impl ContextData {
             .unwrap()
             .send_transaction(webrender_document, txn);
     }
+
+    /// Returns true if presentation id was updated (was newer)
+    fn check_and_update_presentation_id(&mut self, presentation_id: PresentationId) -> bool {
+        if presentation_id > self.current_presentation_id {
+            self.current_presentation_id = presentation_id;
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Returns new presentation id
+    fn next_presentation_id(&mut self) -> PresentationId {
+        let res = PresentationId(self.next_presentation_id.0);
+        self.next_presentation_id.0 += 1;
```

Comment:
in that case we could have special handling in check_and_update (if we have try to update very big with very small), but  given that it will not be problem in practice I do not think it's worth it.